dev-docs/RFCs/v8.1/gpu-accelerated-polygon-clipping.md

```diff
@@ -0,0 +1,153 @@
+
+
+## Proposed Approach
+
+A texture based filtering can be implemented to perform PIP test on GPU. The idea is, user provided polygon(s) is triangulated and rendered to an offline frame buffer. And then, given points are run through transform feedback loop, each point is translated into the texture space and above texture is sampled, result of this sample determines if the point is inside or outside the polygon.
+
+
+## Proposed New Components
+
+### `texturefilter` (ShaderModule)
+
+Provides shader functionality, for transforming input point to texture space and sample the texture to determine if the point is inside or outside the region defined by the texture. When performing polygon clipping, this texture is created by rendering polygons to it.
+
+- Uniforms:
+   - `bboxOrigin` (vec2)  // origin of the bounding box
+   - `bboxSize` (vec2)  // Sizes in X and Y directions of the bounding box
+   - `maskTexture` (sampler2D) // Texture created by rendering polygons
+- VS Function:
+     - `vec3 isWithInTextureMask(vec2 point)`: returns [flag, id, pointIndex]
+
+
+### GPUPolygonFilter (JS Class) (WebGL2 only)
+
+Class that builds a texture mask from given set of polygons, performs transform feedback loop on given set of points and generates the results. It internally uses `texturefilter` shader module.
+
+#### Methods
+
+##### Constructor
+
+Constructor has following arguments, it constructs required objects such as `Texture2D` and `Transform`.
+
+* `gl`(WebGL2Context): WebGL Context
+* `opts` (Optional, Default: {}): Object containing following optional fields.
+* `opts.polygons` (Array, Optional): One or an array of polygon objects. Each polygon object contains following :
+ - `id` (Integer, Optional) : An integer in [0, 255] range, value outside this range will be clamped to this range. When not provided, index of this object in the array used as the value.
+ - `polygon` (Array) : A simple or complex polygon in following format:
+
+  Simple polygon: [[x1, y1], [x2, y2], ...]
+  Complex polygon (with hole) :
+    [                                           
+      [[x1, y1], [x2, y2], ...], // outer ring
+      [[hx1, hy1], [hx2, hy2], ...] // hole
+    ]
+
+##### update(opts)
+
+`opts` object is same as `opts` argument of constructor. Once object is constructed, polygons can be updated.
+
+##### run(opts)
+
+* `opts.positionBuffer` (Buffer) : `Buffer` object containing position data of input points, must contain X and Y values.
+* `opts.count` (Integer) : Count of number of points to be processed.
+* `opts.resultBuffer` (Buffer): `Buffer` object where results will be stored. It should hold at least 3 elements (vec3) for each input point. Once this method is completed this buffer will contain a vec3 [`flag`, `id`, `index`]. Each element (vec3) of this buffer contains the result of corresponding point (vec2) from the `positionBuffer`.
```

Comment:
This can be simplified to an array of scalar floats or ints: `-1` means outside, otherwise index of the polygon it intersected. `index` is redundant with the array index of the entry.